### ISSUE TITLE
[RFC] Allow or disallow `<script>` in content. 

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#5339]
Until now, we've always allowed `<script>` tags in content. 

In fact, more than a few times, people have submitted "serious security issues", because a registered and logged on Editor is able to put `<script>alert('xss');</script>` in a title field. 

Similarly, we allow `<iframe>` in wysiwyg fields, because otherwise you couldn't insert Youtube and Vimeo embeds. However, this means it's also possible to (unknowingly to an editor) insert remote javascripts in your content, because the editor copy/pasted it from a different website. Because CKEditor does _not_ execute these scripts, this can happen without the editor noticing it. 

We will also filter `text`, `textarea` and `html` fields using the `htmlcleaner:    allowed_tags` setting, expanded with the `wysiwyg` settings like `embed`. This would prevent inadvertently adding script to the wysiwyg fields by an editor. 

Items to do: 

- [ ] Use our sanitation for all text-like fields, not just Markdown.
- [ ] Make order consistent: first sanitize, and _then_ parse snippets as twig.
- [ ] When explicitly allowing things like 'embed' in Wysiwig, temporarily add these to whitelist for sanitizing.
- [ ] Document these changes, and in extensions documentation mention that extensions are able to insert stuff _after_ sanitization took place. 
- [ ] Always disallow `<script>` and `onX='..'` attributes in content, because that shit's whack, yo!

**Comment by Puddingboy** [comment]
```
<script>
    (function(){
        var keyevt = "keydown,keyup,keypress".split(',');
        var warningspan = warningelement;
        var content = contentelement;
        var warningmessage = "{{app.content.placeWhereThatMessageWouldBe}}";
        function warnUserIfScriptExist(){
            if(content.innerHTML.match(/<script/i))warningspan.innerHTML = warningmessage;
        }
        for(var key in evt){
            document.addEventListener(key,warnUserIfScriptExist,false);
        }
        // Please note that click events also need listeners for copy+paste by mouseclick
        // furthermore: this code should be placed iteratively so further adaption will be required
    }())
</script>
```
This is a draft. I will edit this later

**Comment by bobdenotter** [comment]
@Puddingboy If we choose to fix this, it should be done on the server-side. Similar to what we already do with Markdown. See here, for example: 

https://github.com/bolt/bolt/blob/release/3.0/src/Legacy/Content.php#L162-L170

**Comment by Puddingboy** [comment]
// like discussed elsewere: my suggestion was meant as an overlay feature for people who wish to disable the filter, but still wish to receive some sort of warning to detect if a script has slipped in by accident. With this I will cease effort on previous suggested draft.

**Comment by SahAssar** [comment]
Btw, I have noticed something odd on one of our customer sites regarding this... They apparently use some of Nortons so called "security products", and those apparently inject an object tag into all iframes (including bolt's editor). When the customer saves the page and someone without Nortons "security products" visits the page on the frontend they get a "missing plugin" warning from the browser since the norton plugin isn't installed.

So if this could act on `<object>` tags aswell as script/iframe tags that'd be great :)

**Comment by bobdenotter** [comment]
Wow, `<object>`.. haven't seen that in a while. :-)

Sure thing, though.

**Comment by bobdenotter** [comment]
Fixed by #5611
